AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.m

```diff
@@ -99,7 +115,44 @@ - (void)channel:(id<MSChannelProtocol>)__unused channel
       }
       target = target.parentTarget;
     }
+
+    // Should the target collect the device Id.
+    target = self.transmissionTarget;
+    while (target) {
+      if (target.propertyConfigurator.shouldCollectDeviceId) {
```

Comment:
We need to talk about this offline. There wouldn't be no inheritance for device ID. Confirmed by PM.